tests/integration/s2n_test_common.py

```diff
@@ -0,0 +1,218 @@
+from s2n_test_constants import *
+from time import sleep
+
+CERT = TEST_ECDSA_CERT
+KEY = TEST_ECDSA_KEY
+
+ALL_VERSIONS = [S2N_SSLv3, S2N_TLS10, S2N_TLS11, S2N_TLS12, S2N_TLS13]
+
+class Cipher():
+    def __init__(self, name, min_version):
+        self.name = name
+        self.min_version = min_version
+
+    def __str__(self):
+        return self.name
+
+ALL_CIPHERS = [
+    Cipher("TLS_AES_256_GCM_SHA384", S2N_TLS13),
+    Cipher("TLS_CHACHA20_POLY1305_SHA256", S2N_TLS13),
+    Cipher("TLS_AES_128_GCM_SHA256", S2N_TLS13)
+]
+
+class Mode(Enum):
+    client = 0
+    server = 1
+
+    def __str__(self):
+        return "Client" if self is Mode.client else "Server"
+
+    def other(self):
+        return Mode.server if self is Mode.client else Mode.client
+
+ALL_MODES = [e.value for e in Mode]
+
+class Status:
+    def __init__(self, name, color):
+        self.name = name
+        self.color = color
+
+    def __str__(self):
+        return with_color(self.name, self.color)
+
+PASSED = Status("PASSED", 32 )
+FAILED = Status("FAILED", 31 )
+
+class Result:
+    def __init__(self, error=None):
+        self.error = error
+        self.status = PASSED if error == None else FAILED
+
+    def __str__(self):
+        result = str(self.status)
+        if self.error:
+            result += "\n\t\t%s %s" % (with_color("Error:", FAILED.color), self.error)
+        return result
+
+class Scenario:
+    def __init__(self, s2n_mode, host, port, version=None, cipher=None, s2n_flags=[], peer_flags=[]):
+        self.s2n_mode = s2n_mode
+        self.host = host
+        self.port = port
+        self.version = version
+        self.cipher = cipher
+        self.s2n_flags = s2n_flags
+        self.peer_flags = peer_flags
+
+    def __str__(self):
+        cipher = self.cipher if self.cipher else "ANY"
+        return "\tMode:%s Endpoint:%s:%s Cipher:%s" % (self.s2n_mode, self.host, self.port, str(cipher).ljust(30))
+
+def with_color(msg, color):
+    if sys.stdout.isatty():
+        return "\033[%d;1m%s\033[0m" % (color, msg)
+    else:
+        return msg
+
+def cleanup_processes(*processes):
+    for p in filter(None, processes):
+        p.kill()
+        p.wait()
+
+def create_thread_pool():
+    threadpool_size = multiprocessing.cpu_count() * 2  # Multiply by 2 since performance improves slightly if CPU has hyperthreading
+    print("\tCreating ThreadPool of size: " + str(threadpool_size))
+    threadpool = ThreadPool(processes=threadpool_size)
+    return threadpool
+
+def get_error(process):
+    return process.stderr.readline().decode("utf-8")
+
+def wait_for_output(process, marker, line_limit=10):
+    for count in range(line_limit):
+        line = process.stdout.readline().decode("utf-8")
+        if marker in line:
+            return True
+    return False
+
+def get_process(cmd):
+    return subprocess.Popen(cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+def get_test(get_peer, test_func=None):
```

Comment:
Should this parameter be `get_peer` or `get_process`? A comment may help here as well. I think I know how to add another test using `get_test()` but I'm not positive.